apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/TableEditor/RLSManagement/RLSManagement.tsx

```diff
@@ -0,0 +1,401 @@
+              <div className="flex items-center gap-2">
+                <Button
+                  type="default"
+                  size="tiny"
+                  onClick={handleGeneratePolicies}
+                  loading={isGenerating}
+                  disabled={isGenerating || !isRlsEnabled}
+                >
+                  {isGenerating
+                    ? 'Generating policies...'
+                    : generateFailed
+                      ? 'Try generating again'
+                      : 'Generate policies'}
+                </Button>
+                {showPermissionButton && (
+                  <Button
+                    type="default"
+                    size="tiny"
+                    onClick={() => setIsOptInModalOpen(true)}
+                    disabled={isGenerating}
+                  >
+                    Permission settings
+                  </Button>
+                )}
+              </div>
+            </div>
+          </CardContent>
+        )
+      }
+      return (
+        <CardContent>
+          <p className="text-sm text-foreground-lighter">No policies exist for this table</p>
+        </CardContent>
+      )
+    }
+
+    const handleRemoveGeneratedPolicy = (index: number) => {
+      // Find the index in generatedPoliciesList that corresponds to the allPoliciesList index
+      const generatedIndex = index - existingPoliciesList.length
+      if (generatedIndex >= 0 && generatedIndex < generatedPolicies.length) {
+        const updatedPolicies = generatedPolicies.filter((_, i) => i !== generatedIndex)
+        onGeneratedPoliciesChange?.(updatedPolicies)
+
+        // Track policy removal
+        if (project?.ref) {
+          track('rls_generated_policy_removed')
+        }
+      }
+    }
+
+    return (
+      <PolicyList
+        policies={allPoliciesList}
+        className="border-0 rounded-none"
+        onRemove={isNewRecord && !isDuplicating ? handleRemoveGeneratedPolicy : undefined}
+      />
+    )
+  }
+
+  if (!project) return null
+
+  if (isNewRecord && !isDuplicating) {
+    return (
+      <div>
+        <div className="flex items-center mb-4 gap-2">
+          <div className="flex-1">
+            <h3>Policies</h3>
+            <p className="text-sm text-foreground-lighter">
+              Set rules around who can read and write data to this table
+            </p>
+          </div>
+        </div>
+
+        <Card>{renderPolicies()}</Card>
+        {generatedPolicies?.length > 0 && (
+          <p className="text-xs text-foreground-lighter mt-4 flex items-center gap-2">
+            <InfoIcon size={16} strokeWidth={1.5} />
+            Review the policies generated before creating your table
+          </p>
+        )}
+        <AIOptInModal visible={isOptInModalOpen} onCancel={() => setIsOptInModalOpen(false)} />
+      </div>
+    )
+  }
+
+  return (
+    <div>
+      <div className="flex items-center mb-4 gap-2">
+        <div className="flex-1">
+          <h3>Policies</h3>
```

Comment:
nit: for consistency based on the other sections in the table editor, could use h5 instead